### PR TITLE
Add CirclCI, TravisCI, and Appveyor configs

### PR DESCRIPTION
**Pull request by SpainTrain**
[Review on Reviewable](https://reviewable.io/reviews/atomlinter/linter-markdown/14)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+branches:
+  only:
+    - master
+notifications:
+  email:
+    on_success: never
+    on_failure: change
+os: osx
+
+script: apm install && npm tst
```
Comment (Arcanemagus) [inline .travis.yml]: I'd put this as `test`, just to match the actual name in the `package.json`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+sudo: true
```
Comment (Arcanemagus) [inline .travis.yml]: This is meaningless on the OS X platform.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  - appveyor DownloadFile https://atom.io/download/windows -FileName AtomSetup.exe
+  - AtomSetup.exe /silent
+
+build_script:
+  - cd %APPVEYOR_BUILD_FOLDER%
+  - SET PATH=%LOCALAPPDATA%\atom\bin;%PATH%
+  - apm clean
+  - apm install
+  # TODO: add specs (#13)
+  - npm tst
```
Comment (Arcanemagus) [inline appveyor.yml]: Same thing, I'd put it as `npm test` myself.

I guess it doesn't really matter though.

**Comment by SpainTrain** [comment]
:white_check_mark:

**Comment by kepler0** [comment]
The Travis script is out of date. The Atom team has published an updated one which supports OS X and Linux, testing on the Stable Atom and Beta Atom.

**Comment by Arcanemagus** [comment]
@kepler0 The only real differences between the current one there and this is the addition of testing Linux on Travis-CI and the beta channel testing. Since CircleCI does Linux testing faster (and nicer looking), there really isn't much point in that, although the beta channel testing is nice.

On the other hand the OS X builds take up to 2x as long as the Linux builds so there isn't much reason _not_ to also test Linux on Travis :stuck_out_tongue:.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+branches:
+  only:
+    - master
+notifications:
+  email:
+    on_success: never
+    on_failure: change
+os: osx
+
+script: apm install && npm test
```
Comment (Arcanemagus) [inline .travis.yml]: This is failing for a few different reasons, the primary being you never installed Atom :stuck_out_tongue:.

**Comment by SpainTrain** [comment]
Ya, I am happy to use whatever travis script.  I just copied this one from linter-eslint.

Relatedly, with some automation around https://github.com/AtomLinter/Meta/issues/7 we could apply such updates to all "subscribed" linter repos at once.

**Comment by SpainTrain** [comment]
Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/atomlinter/linter-markdown/14)*

**Pull request merged**